src/layman/upgrade/upgrade_v2_0_util.py

```diff
@@ -62,5 +62,9 @@ def get_layer_input_files(workspace, layername):
 
 
 def safe_delete(path):
-    if os.path.exists(path):
+    try:
         os.rmdir(path)
+    except FileNotFoundError:
+        pass
+    except OSError:
+        logger.warning(f"    Directory {path} was not deleted, because it is not empty.")
```

Comment:
I prefer to log such warning only in case of three root directories
- {settings.LAYMAN_QGIS_DATA_DIR}/workspaces
- {settings.LAYMAN_NORMALIZED_RASTER_DATA_DIR}/workspaces
- {settings.LAYMAN_DATA_DIR}/workspaces

otherwise there can be a lot of warnings that won't help much.

Reply:
Incorporated.